float32[] angles
float32[] pose